src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,78 @@
+body {
+  font-family: "Arial", sans-serif;
+}
+  
+.calendar {
+  border: 1px solid red;
+  margin: 100px auto;
+  width: $cell-size * 7 + 8;
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+  
+  &__days-cell {
+    border: 1px solid black;
+    width: $cell-size;
+    height: $cell-size;
+    background: #eee;
+    text-align: center;
+    font-size: 30px;
+    line-height: $cell-size;
+    transition: all linear 0.5s;
+    cursor: pointer;
+    @for $index from 1 to 31+1 {
+      &:nth-child(#{$index}n) {
+        display: inline-block;
+        &:before {
+          content: "#{$index}";
+        }
+      }
+    }
+
+    &:hover {
+      position: relative;
+      top: -20px;
+      background: pink;
+      color: #fff;
+    }
+  }
+
+  &--month-length_28 {
+    .calendar__days-cell {
+      @for $index from 29 to 32 {
+        &:nth-child(#{$index}) {
+          display: none;
+        }
+      }
+    }
+  }
+
+  &--month-length_29 {
+    .calendar__days-cell {
+      @for $index from 30 to 32 {
+        &:nth-child(#{$index}) {
+          display: none;
+        }
+      }
+    }
+  }
+
+  &--month-length_30 {
+    .calendar__days-cell {
+      &:nth-child(31n) {
+        display: none;
+      }
+    }
```

Comment:
let's try to use one loop for creating these modifiers instead of duplicates code